modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridNearAtomicAbstractUpdateFuture.java

```diff
@@ -401,7 +401,7 @@ final void onPrimaryError(GridNearAtomicAbstractUpdateRequest req, GridNearAtomi
         Collection<Object> keys = new ArrayList<>(keys0.size());
 
         for (KeyCacheObject key : keys0)
-            keys.add(cctx.cacheObjectContext().unwrapBinaryIfNeeded(key, keepBinary, false));
+            keys.add(cctx.cacheObjectContext().unwrapBinaryIfNeeded(key, true, false));
```

Comment:
I suggest `keys.add(cctx.toCacheKeyObject(key));` instead

Reply:
@alamar , thanks for the review. This is a good suggestion. Since we already have a collection of KeyCacheObject probably it makes sense to put that as a parameter of the exception `err.add(keys,... )` (all keys will be copied into `CachePartialUpdateCheckedException.failedKeys`). What do you think?